sdks/python/apache_beam/io/external/kafka.py

```diff
@@ -118,35 +118,112 @@ def expand(self, pbegin):
             payload.SerializeToString(),
             self.expansion_service))
 
-  @staticmethod
-  def _encode_map(dict_obj):
-    kv_list = [(key.encode('utf-8'), val.encode('utf-8'))
-               for key, val in dict_obj.items()]
-    coder = IterableCoder(TupleCoder(
-        [LengthPrefixCoder(BytesCoder()), LengthPrefixCoder(BytesCoder())]))
-    coder_urns = ['beam:coder:iterable:v1',
-                  'beam:coder:kv:v1',
-                  'beam:coder:bytes:v1',
-                  'beam:coder:bytes:v1']
-    return ConfigValue(
-        coder_urn=coder_urns,
-        payload=coder.encode(kv_list))
-
-  @staticmethod
-  def _encode_list(list_obj):
-    encoded_list = [val.encode('utf-8') for val in list_obj]
-    coder = IterableCoder(LengthPrefixCoder(BytesCoder()))
-    coder_urns = ['beam:coder:iterable:v1',
-                  'beam:coder:bytes:v1']
-    return ConfigValue(
-        coder_urn=coder_urns,
-        payload=coder.encode(encoded_list))
-
-  @staticmethod
-  def _encode_str(str_obj):
-    encoded_str = str_obj.encode('utf-8')
-    coder = LengthPrefixCoder(BytesCoder())
-    coder_urns = ['beam:coder:bytes:v1']
-    return ConfigValue(
-        coder_urn=coder_urns,
-        payload=coder.encode(encoded_str))
+
+class WriteToKafka(ptransform.PTransform):
+  """
+    An external PTransform which writes KV data to a specified Kafka topic.
+    If no Kafka Serializer for key/value is provided, then key/value are
+    assumed to be byte arrays.
+
+    Note: To use this transform, you need to start the Java expansion service.
+    Please refer to the portability documentation on how to do that. The
+    expansion service address has to be provided when instantiating this
+    transform. During pipeline translation this transform will be replaced by
+    the Java SDK's KafkaIO.
```

Comment:
Prob. move this information to the top of the module (instead of repeating for read and write transforms).

Reply:
Wasn't sure how this is displayed in IDEs for Python development but seems fair not to repeat it.